src/main/java/co/rsk/federate/watcher/FederationWatcherListenerImpl.java

```diff
@@ -0,0 +1,71 @@
+package co.rsk.federate.watcher;
+
+import co.rsk.federate.BtcToRskClient;
+import co.rsk.federate.btcreleaseclient.BtcReleaseClient;
+import co.rsk.peg.federation.Federation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.Objects;
+
+public class FederationWatcherListenerImpl implements FederationWatcherListener {
```

Comment:
Shouldn't this class have tests as well?

Reply:
Added unit tests, let me know if it looks good